src/core/stdc/stdarg.d

```diff
@@ -459,10 +459,19 @@ else version (X86_64)
     {
     }
 
+    import core.stdc.stdlib : alloca;
+
     ///
-    void va_copy(out va_list dest, va_list src)
+    void va_copy(out va_list dest, va_list src, void* storage = alloca(__va_list_tag.sizeof))
```

Comment:
> What's worrisome here is - this is a neat trick but could break easily. alloca is finicky and I'm note sure we have enough guarantees at the language definition level to ensure the call goes onto the callee stack.

The thing is, currently `va_copy` never works on posix 64-bit.  The alloca arg hack works.  I don't know what alloca could ever do apart from allocating in the caller's stack frame.

> Unrelated tactical suggestion: (*dest).sizeof is more self-explanatory.

I'm really not sure it is.  I don't want anyone looking at this code and thinking they know what it does.  The other `va_*` functions are rather nasty, and forcing any would-be modifier to go and look at the actual definitions is a good thing.

> also s/out/ref/ for efficiency?

Then it wouldn't match the signatures for the other platforms.